### PR TITLE
(fix) O3-3136: Appointments: Don't Allow Checking In to Cancelled App…

### PR DESCRIPTION
**Pull request by mogoodrich**
…ointments

## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

### CODE REVIEW

**Review by chibongho — APPROVED** [review]
Looks fine to me. I would have put the isCancelled case first to make the condition for other cases less complex.

File: packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx [hunk 1]
```diff
@@ -47,21 +48,27 @@ const AppointmentsActions: React.FC<AppointmentsActionsProps> = ({ appointment }
...
-      case checkInButton.enabled && !hasActiveVisitToday && isTodaysAppointment: {
+      case checkInButton.enabled && !hasActiveVisitToday && isTodaysAppointment && !isCancelled: {
         return <CheckInButton patientUuid={patientUuid} appointment={appointment} />;
       }
+      case isCancelled:
+        return (
+          <Button kind="danger--ghost" iconDescription={t('cancelled', 'Cancelled')} size="sm">
+            {t('cancelled', 'Cancelled')}
+          </Button>
+        );
```
Comment (mseaton) [inline packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx:71]: It's getting a little hard to follow.  Wouldn't it be easier and clearer if this were just a series of if then return statements, and at the top putting the Cancelled case?

```if (isCancelled) return cancel button```

Then everything below this can proceed without having to worry about the Cancelled case...

**Review by mseaton — APPROVED** [review]
I'm sure it's fine, but see my comment, might be clearer...

File: packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx (hunk 1 above)
Comment (mogoodrich) [reply packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx:71]: Both you and @chibongho commented pretty much the same thing, so I'll switch I guess (we don't need to change to and if/then, fwiw, just put the Cancel as the first switch statement)

FWIW, I did have cancel first originally, but thought there was value in explicitly doing the logic for each case, even if it was repetitive, in case someone unknowing moved things around... (but I think you both have convinced me)

**Comment by mogoodrich** [comment]
Thanks @mseaton  and @chibongho , I made the change you are suggesting, merging in now...
